generators/wp/templates/chisel-starter-theme/Chisel/Media.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Chisel;
+
+/**
+ * Class Media
+ * @package Chisel
+ *
+ * Default media settings for Chisel
+ */
+class Media {
+
+	public function __construct() {
+		// $this->addImageSizes();
+		// add_action( 'after_setup_theme', array( $this, 'defaultMediaSetting' ) );
+		// add_filter( 'image_size_names_choose', array( $this, 'customImageSizes' ) );
+		add_action( 'upload_mimes', array( $this, 'customMimeTypes') );
+		add_action( 'jpeg_quality', array( $this, 'customJpegQuality') );
+		add_filter( 'embed_oembed_html', array( $this, 'customOembedFilter' ), 10, 4);
+	}
+
+	/**
+	 * Add various image sizes
+	 */
+	public function addImageSizes() {
+		add_image_size( 'small', 225, 9999 );
+		add_image_size( '800w', 800, 9999 );
+	}
+
+	/**
+	 * Add custom image sizes option to WP admin
+	 * @param  array $sizes Default sizes
+	 * @return array        Updated sizes
+	 */
+	public function customImageSizes( $sizes ) {
+		return array_merge( $sizes, array(
+			'small' => __( 'Small' ),
+		));
+	}
+
+	/**
+	 * Default settings when adding or editing post images
+	 */
+	public function defaultMediaSetting() {
+		update_option( 'image_default_align', 'center' );
+		update_option( 'image_default_link_type', 'none' );
+		update_option( 'image_default_size', 'full' );
+	}
+
+	/**
+	 * Allowing additional image types
+	 * @param  array $mimes mime types
+	 * @return array        extended mime types
+	 */
+	function customMimeTypes( $mimes ) {
+		$mimes['svg'] = 'image/svg+xml';
```

Comment:
Do we really need to make it enabled by default? I believe not only wordpress but also some plugins that allow users to upload their files use this validation and this will lead to almost uncontrolled uploading of svgs, that are really vulnerable to script injections.

If we want to enable it by default we should definitely make it more secure. Maybe https://github.com/darylldoyle/svg-sanitizer will help?

Above library also allows to minify uploaded svgs

Reply:
Thanks for pointing this out, it's been while since I was looking at SVG support in WP and now I see this is not recommended practice. Maybe we could just recommend in development guidelines to use this plugin https://wordpress.org/plugins/safe-svg/ as it's from the same author you mentioned.

Reply:
I've seen the plugin, but it probably has some paid features and definitely add some entries in the dashboard. I don't think we need such a plugin and we could just write a simple scan using this free (GPL v2) library. But if you want to introduce another plugin, I'm ok with that ;)

Reply:
I would just mention it in guidelines for now as a recommend practice, please can use the plugin only if they really need. I agree having this implemented in Chisel would be better at some point, but I guess that to get that library to Chisel we would have to use Composer and setting that all up is probably more work.

Reply:
We can use composer or just load the library to Chisel and update it before each release. I'm afraid not much devs read the readme from beginning to the end, so there will be situations that one will enable svg mime type, but won't even know how insecure it is. We've already have a case with a malware installed on wordpress, and this can be REALLY painful to remove completely.

Reply:
I mean this is my concern as long as we are using it in xfive and some bad things happen.